### ISSUE TITLE
[BUG]: Logging.info() not stored in server logs

### ISSUE THREAD

**Issue by patricksavill** [zenml-io/zenml#1690]
### Contact Details [Optional]

pjsavill@gmail.com

### System Information

ZENML_LOCAL_VERSION: 0.41.0
ZENML_SERVER_VERSION: 0.41.0
ZENML_SERVER_DATABASE: sqlite
ZENML_SERVER_DEPLOYMENT_TYPE: other
ZENML_CONFIG_DIR: /home/patrick/.config/zenml
ZENML_LOCAL_STORE_DIR: /home/patrick/.config/zenml/local_stores
ZENML_SERVER_URL: http://192.168.1.27:8080
ZENML_ACTIVE_REPOSITORY_ROOT: None
PYTHON_VERSION: 3.8.10
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'linux', 'linux_distro': 'ubuntu', 'linux_distro_like': 
'debian', 'linux_distro_version': '20.04'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: default
ACTIVE_USER: Patrick
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: b6503272-84f0-447e-a046-2e61511fcc3b
ANALYTICS_USER_ID: 6cd93a1e-e7dd-4ec5-b9f5-0ff625eb1514
ANALYTICS_SERVER_ID: fd311155-da3a-4748-9d26-c3d6445c99b3
INTEGRATIONS: ['kaniko', 'scipy']
PACKAGES: {'zipp': '3.16.1', 'zenml': '0.41.0', 'widgetsnbextension': '3.6.4', 
'websockets': '11.0.3', 'websocket-client': '1.6.1', 'webencodings': '0.5.1', 
'wcwidth': '0.2.6', 'watchgod': '0.8.2', 'validators': '0.18.2', 'uvloop': 
'0.17.0', 'uvicorn': '0.17.6', 'urllib3': '2.0.3', 'tzdata': '2023.3', 'typing':
'3.7.4.3', 'typing-extensions': '4.7.1', 'traitlets': '5.9.0', 'tornado': 
'6.3.2', 'tinycss2': '1.2.1', 'threadpoolctl': '3.2.0', 'terminado': '0.17.1', 
'starlette': '0.17.1', 'stack-data': '0.6.2', 'sqlmodel': '0.0.8', 
'sqlalchemy2-stubs': '0.0.2a35', 'sqlalchemy': '1.4.41', 'sqlalchemy-utils': 
'0.38.3', 'soupsieve': '2.4.1', 'sniffio': '1.3.0', 'smmap': '5.0.0', 'sklearn':
'0.0.post5', 'send2trash': '1.8.2', 'scipy': '1.10.1', 'scikit-learn': '1.3.0', 
'rpds-py': '0.8.10', 'rich': '12.6.0', 'rfc3986-validator': '0.1.1', 
'rfc3339-validator': '0.1.4', 'requests': '2.31.0', 'referencing': '0.29.1', 
'pyzmq': '25.1.0', 'pyyaml': '5.4.1', 'pytz': '2023.3', 'python-terraform': 
'0.10.1', 'python-multipart': '0.0.6', 'python-json-logger': '2.0.7', 
'python-jose': '3.3.0', 'python-dotenv': '1.0.0', 'python-dateutil': '2.8.2', 
'pyparsing': '2.4.7', 'pymysql': '1.0.3', 'pygments': '2.15.1', 'pydantic': 
'1.10.11', 'pycparser': '2.21', 'pure-eval': '0.2.2', 'ptyprocess': '0.7.0', 
'prompt-toolkit': '3.0.39', 'prometheus-client': '0.17.1', 'platformdirs': 
'3.8.1', 'pkgutil-resolve-name': '1.3.10', 'pickleshare': '0.7.5', 'passlib': 
'1.7.4', 'parso': '0.8.3', 'pandocfilters': '1.5.0', 'pandas': '2.0.3', 
'packaging': '23.1', 'overrides': '7.3.1', 'orjson': '3.8.14', 'openexr': 
'1.3.9', 'numpy': '1.24.3', 'notebook': '6.5.4', 'notebook-shim': '0.2.3', 
'nest-asyncio': '1.5.6', 'nbformat': '5.9.1', 'nbconvert': '7.6.0', 'nbclient': 
'0.8.0', 'nbclassic': '1.0.0', 'mistune': '3.0.1', 'matplotlib': '3.7.2', 
'matplotlib-inline': '0.1.6', 'markupsafe': '2.1.3', 'kiwisolver': '1.4.4', 
'jupyterlab-widgets': '1.1.4', 'jupyterlab-pygments': '0.2.2', 'jupyter-server':
'2.7.0', 'jupyter-server-terminals': '0.4.4', 'jupyter-events': '0.6.3', 
'jupyter-core': '5.3.1', 'jupyter-client': '8.3.0', 'jsonschema': '4.18.3', 
'jsonschema-specifications': '2023.6.1', 'joblib': '1.3.1', 'jinja2': '3.1.2', 
'jedi': '0.18.2', 'isodate': '0.6.1', 'ipywidgets': '7.7.5', 'ipython': 
'8.12.2', 'ipython-genutils': '0.2.0', 'ipykernel': '6.24.0', 
'importlib-resources': '6.0.0', 'importlib-metadata': '6.8.0', 'httptools': 
'0.6.0', 'httplib2': '0.19.1', 'h11': '0.14.0', 'greenlet': '2.0.2', 
'gitpython': '3.1.32', 'gitdb': '4.0.10', 'fonttools': '4.41.0', 
'fastjsonschema': '2.17.1', 'fastapi': '0.75.2', 'fastapi-utils': '0.2.1', 
'executing': '1.2.0', 'exceptiongroup': '1.1.2', 'ecdsa': '0.18.0', 'docker': 
'6.1.3', 'distro': '1.8.0', 'defusedxml': '0.7.1', 'decorator': '5.1.1', 
'debugpy': '1.6.7', 'cycler': '0.11.0', 'cryptography': '41.0.2', 'contourpy': 
'1.1.0', 'commonmark': '0.9.1', 'comm': '0.1.3', 'cloudpickle': '2.2.1', 
'click': '8.1.5', 'click-params': '0.3.0', 'charset-normalizer': '3.2.0', 
'cffi': '1.15.1', 'bleach': '6.0.0', 'beautifulsoup4': '4.12.2', 'backoff': 
'1.10.0', 'backcall': '0.2.0', 'azure-mgmt-resource': '23.0.1', 
'azure-mgmt-core': '1.4.0', 'azure-core': '1.28.0', 'azure-common': '1.1.28', 
'attrs': '23.1.0', 'asttokens': '2.2.1', 'asgiref': '3.7.2', 'argon2-cffi': 
'21.3.0', 'argon2-cffi-bindings': '21.2.0', 'anyio': '3.7.1', 
'analytics-python': '1.4.post1', 'alembic': '1.8.1', 'xkit': '0.0.0', 'wheel': 
'0.34.2', 'wadllib': '1.3.3', 'usb-creator': '0.3.7', 'unattended-upgrades': 
'0.1', 'ufw': '0.36', 'ubuntu-drivers-common': '0.0.0', 
'ubuntu-advantage-tools': '8001', 'systemd-python': '234', 'ssh-import-id': 
'5.10', 'six': '1.14.0', 'sip': '4.19.21', 'simplejson': '3.16.0', 'setuptools':
'45.2.0', 'secretstorage': '2.3.1', 's3transfer': '0.3.3', 'rsa': '4.0', 
'roman': '2.0.0', 'requests-unixsocket': '0.2.0', 'reportlab': '3.5.34', 'qtpy':
'1.9.0', 'pyxdg': '0.26', 'python-debian': '0.1.36ubuntu1', 'python-apt': 
'2.0.1+ubuntu0.20.4.1', 'pyrfc3339': '1.1', 'pyqt5': '5.14.1', 'pynacl': 
'1.3.0', 'pymacaroons': '0.13.0', 'pyjwt': '1.7.1', 'pygobject': '3.36.0', 
'pycups': '1.9.73', 'pycrypto': '2.6.1', 'pycairo': '1.16.2', 'pyasn1': '0.4.2',
'psutil': '5.5.1', 'protobuf': '3.6.1', 'pip': '20.0.2', 'pillow': '7.0.0', 
'pexpect': '4.6.0', 'pbr': '5.4.5', 'paramiko': '2.6.0', 'olefile': '0.46', 
'oauthlib': '3.1.0', 'netifaces': '0.10.4', 'monotonic': '1.5', 'mock': '3.0.5',
'mako': '1.1.0', 'macaroonbakery': '1.3.1', 'louis': '3.12.0', 'lockfile': 
'0.12.2', 'lazr.uri': '1.0.3', 'lazr.restfulclient': '0.14.2', 'launchpadlib': 
'1.10.13', 'language-selector': '0.1', 'keyring': '18.0.1', 'jmespath': '0.9.4',
'idna': '2.8', 'gpg': '1.13.1-unknown', 'future': '0.18.2', 'fasteners': 
'0.14.1', 'entrypoints': '0.3', 'duplicity': '0.8.12.0', 'docutils': '0.16', 
'distro-info': '0.23ubuntu1', 'defer': '1.0.6', 'dbus-python': '1.2.16', 
'cupshelpers': '1.0', 'configobj': '5.0.6', 'command-not-found': '0.3', 
'colorama': '0.4.3', 'chardet': '3.0.4', 'certifi': '2019.11.28', 'brlapi': 
'0.7.0', 'botocore': '1.16.19', 'blinker': '1.4', 'bcrypt': '3.1.7', 'awscli': 
'1.18.69', 'apturl': '0.5.2', 'apt-xapian-index': '0.49'}

CURRENT STACK

Name: default
ID: 02aa349b-5ee9-4eae-ad70-3b277647a683
Shared: No
User: Patrick / 6cd93a1e-e7dd-4ec5-b9f5-0ff625eb1514
Workspace: default / bf130e5c-bb64-4f72-8533-6b47211b5a01

ORCHESTRATOR: default

Name: default
ID: be2095b1-7e60-4828-a33d-7b915f891a53
Type: orchestrator
Flavor: local
Configuration: {}
Shared: No
User: Patrick / 6cd93a1e-e7dd-4ec5-b9f5-0ff625eb1514
Workspace: default / bf130e5c-bb64-4f72-8533-6b47211b5a01

ARTIFACT_STORE: default

Name: default
ID: 2e343cb9-fe1a-48c0-aa7d-eacd841d917d
Type: artifact_store
Flavor: local
Configuration: {'path': ''}
Shared: No
User: Patrick / 6cd93a1e-e7dd-4ec5-b9f5-0ff625eb1514
Workspace: default / bf130e5c-bb64-4f72-8533-6b47211b5a01

### What happened?

When using `logging.info` in a @step function the resulting logs, whilst printed to the console output, are not shown in the pipeline run in the server.

I expected to see the logs that are shown on the console in the dashboard.
I'm running ZenML on the default stack, and have the 0.41.0 docker image running as the server.

Here is the code, that does almost nothing (attached as a text file)
[hello-zml.txt](https://github.com/zenml-io/zenml/files/12065053/hello-zml.txt)

And the offending lack of logs:
![image](https://github.com/zenml-io/zenml/assets/11937983/aec065cb-ab87-488a-a61b-6273a206dd88)
![image](https://github.com/zenml-io/zenml/assets/11937983/92dd3257-6264-495d-a855-67f9d680b2a7)

### Reproduction steps

1. Start up the server instance
2. `zenml connect ...`  to the dockerised server
3. `zenml status` to ensure a connection is up
4. run the python code

### Relevant log output

```shell
Running without an active repository root.
Reusing registered pipeline logging_pipeline (version: 3).
Running pipeline logging_pipeline on stack default (caching enabled)
Step logging_one has started.
INFO:root:Step one, log one
INFO:root:Hello world
Step logging_one has finished in 0.997s.
Step logging_two has started.
INFO:root:Step two
Step logging_two has finished in 1.027s.
Step logging_three has started.
INFO:root:Step three
Step logging_three has finished in 1.639s.
Pipeline run logging_pipeline-2023_07_16-20_43_30_075477 has finished in 2.411s.
Dashboard URL: http://192.168.1.27:8080/workspaces/default/pipelines/9e15f412-c791-4eb1-b19a-f58938535f31/runs/1aa3adc7-47e0-4a8d-b79d-1a3be1659e93/dag
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bcdurak** [comment]
A quick update on this issue: [The PR that refactors our logging module](https://github.com/zenml-io/zenml/pull/1733) is out now. With the changes included in this PR, this issue should be resolved now as well.

**Comment by bcdurak** [comment]
The PR is merged and the changes will be available with the next release of ZenML. Closing this issue now.
